forward/forwarder.go

```diff
@@ -109,15 +115,64 @@ func NewForwarder(s Sender, ch shared.SubChannel, logger log.Logger) *Forwarder
 	}
 }
 
+func (f *Forwarder) emit(event interface{}) {
+	for _, listener := range f.listeners {
+		go listener.HandleEvent(event)
```

Comment:
yea, think we should be careful about sneaky concurrency in our application. not to say that this is wrong, but depending upon the number of listeners (should be very small) and the rate of updates, we may need to be more careful about where we gorountine and where we don't. I don't see an any immediate need for this being async.